### PR TITLE
fix: update active subroutine if belongs to rolled fork

### LINKED ISSUE #2723: `rollFork()` has only partial affect
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (6cd6618 2022-08-11T00:03:54.64793456Z)

### What command(s) is the bug in?

_No response_

### Operating System

Linux

### Describe the bug

Calling `vm.rollFork()` correctly updates `block.number` and `block.timestamp` (see also https://github.com/foundry-rs/foundry/issues/2623). However, the ETH balance of an address is *not* affected by `vm.rollFork()`:
```solidity
function testRollFork() public {
	address coinbase = 0x0193d941b50d91BE6567c7eE1C0Fe7AF498b4137;

	vm.createSelectFork(RPC_ENDPOINT, 9);

	assertEq(block.number, 9);
	assertEq(coinbase.balance, 11250000000000000000);

	vm.rollFork(10);
	//vm.createSelectFork(RPC_ENDPOINT, 10); // Test succeeds when using this

	assertEq(block.number, 10); // Passes
	assertEq(coinbase.balance, 16250000000000000000); // Fails, because coinbase.balance has not changed
}
```
The last assertion fails because `coinbase.balance` is not affected by `vm.rollFork(10)`. However, when using `vm.createSelectFork(RPC_ENDPOINT, 10)`. I think both `vm.rollFork(10)` and `vm.createSelectFork(RPC_ENDPOINT, 10)` should behave equivalently in this case.

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
Closes #2723

## Solution
when the rolled fork is the active fork, update the active subroutine right away, this will ensure cached state (from previous block) is discarded

### CODE REVIEW

File: testdata/repros/Issue2723.t.sol [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+        vm.createSelectFork("rpcAlias", 9);
+
+        assertEq(block.number, 9);
+        assertEq(coinbase.balance, 11250000000000000000);
+
+        vm.rollFork(10);
+
+//        assertEq(block.number, 10);
+//        assertEq(coinbase.balance, 16250000000000000000);
```
Comment (onbjerg) [inline testdata/repros/Issue2723.t.sol:22]: These are supposed to not be commented, right?

Comment (mattsse) [reply testdata/repros/Issue2723.t.sol:22]: whoops, should be included, fixing

**Comment by mattsse** [comment]
updated

**Review by onbjerg — APPROVED** [review]
lgtm, merge when test pass

**Pull request merged**